lib/liquid/standardfilters.rb

```diff
@@ -143,11 +143,19 @@ def sort_natural(input, property = nil)
       ary = InputIterator.new(input)
 
       if property.nil?
-        ary.sort { |a, b| a.casecmp(b) }
+        ary.sort { |a, b| a.to_s.casecmp(b.to_s) }
       elsif ary.empty? # The next two cases assume a non-empty array.
         []
       elsif ary.first.respond_to?(:[]) && !ary.first[property].nil?
-        ary.sort { |a, b| a[property].casecmp(b[property]) }
+        ary.sort do |a, b|
+          a = a[property]
+          b = b[property]
+          if a && b
```

Comment:
Would it make sense remove the conditional here and `to_s` nil as well?

Reply:
The conditional pushes nil to the end rather than i beginning similar to how `sort` behaves.

Reply:
That's inconsistent with the `if property.nil?` branch